src/SonarScanner.MSBuild.Shim/ScannerEngineInputGenerator.cs

```diff
@@ -173,11 +173,34 @@ internal bool GenerateProperties(AnalysisProperties analysisProperties, ProjectD
         }
         legacyWriter.WriteGlobalSettings(analysisProperties);
 
-        var sensitiveArgsFromSettingsFile = analysisConfig.AnalysisSettings(false, runtime.Logger).GetAllProperties().Where(x => x.ContainsSensitiveData());
-        engineInput.AddUserSettings(new AggregatePropertiesProvider(cmdLineArgs, new ListPropertiesProvider(sensitiveArgsFromSettingsFile), new ListPropertiesProvider(analysisProperties)));
+        // TODO: We cannot write mapped values to the AnalysisConfig file as it changes behaviour of the scanner-cli, this should be fixed in SCAN4NET-721
+        var mappedArgs = MapDeprecatedArgs(new AggregatePropertiesProvider(
+            cmdLineArgs,
+            new ListPropertiesProvider(analysisConfig.ScannerOptsSettings.Select(x => new Property(x.Id, x.Value.Trim('"')))),
+            new ListPropertiesProvider(analysisConfig.AnalysisSettings(false, runtime.Logger).GetAllProperties().Where(x => x.ContainsSensitiveData())),
+            new ListPropertiesProvider(analysisProperties)));
```

Comment:
I would extract all of these three sources to a local variable, like with `cmdLineArgs`, to help the reader understand what each source is about.